python/GafferImageUI/CatalogueUI.py

```diff
@@ -51,6 +53,78 @@
 ## then this will probably force us to sort this out. For now, is at least
 ## contained within just this file...
 
+##########################################################################
+# Column Configuration
+#
+# All of the catalogs columns are configurable. Columns are identified by
+# a unique key along with a title, value provider and type. Gaffer
+# provides a default set of columns that can be extended or replaced via
+# the registration of new keys or re-registration of existing ones.
+#
+# The defaultColumns variable holds a list of standard columns for all new
+# Catalogue instances that can be modified to affect all Catalogues.
+# Individual catalogues can have override this list of configured columns via
+# the "catalogue:columns" [StringVectorData] metadata entry.
+#
+# Value providers for Standard type columns return a string-coercible, those
+# for Icon type columns return the name of an image on Gaffer's image paths.
+# They are called with two arguments:
+#
+#  - image : A GafferImage.Catalogue.Image plug (not to be confused with
+#      a standard ImagePlug).
+#
+#  - catalogue: The GafferImage.Catalogue instance attached to the UI.
+#
+#  A suitable context is scoped around the call such that catalogue["out"] will
+#  provide the image for the row the value is being generated for, rather than
+#  the user's current selection.
+##########################################################################
+
+__registeredColumns = {}
+
+ColumnType = IECore.Enum.create( "Standard", "Icon" )
+ColumnDefinition = namedtuple( "ColumnDefinition", ( "title", "valueProvider", "type" ) )
+
+def registerColumn( key, title, valueProvider, type = ColumnType.Standard ) :
+
+	definition = ColumnDefinition( title, valueProvider, type )
+	__registeredColumns[ key ] = definition
+
+def deregisterColumn( key ) :
+
+	if key in __registeredColumns :
+		del __registeredColumns[ key ]
+
+def columnDefinition( key ) :
+
+	return __registeredColumns.get( key, None )
+
+def registeredColumns() :
+
+	return __registeredColumns.keys()
+
+#
+# Default columns
```

Comment:
I think this sort of default registration is normally handled in startup configs for particular apps or modules

Reply:
Cool, didn't know if it made sense here as the UI is totally useless without any columns?

Reply:
I'll defer to John on that one. A lot of things are considerably less useful without the startup files.